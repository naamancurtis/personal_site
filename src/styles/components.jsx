import styled from 'styled-components';
import { convertRgbStrToObject, pSBC } from './utility.functions';

export const Line = styled.div`
  width: 100%;
  border-bottom: 3px solid ${({ theme }) => theme.body};
  margin-top: 1rem;
  margin-bottom: 1rem;
  box-shadow: ${(props) => {
    const primaryRgb = convertRgbStrToObject(pSBC(-0.3, props.theme.body, 'c'));
    return `
        2px 2px 8px 0 rgba(${primaryRgb.r}, ${primaryRgb.g}, ${primaryRgb.b}, 1);
        -2px -2px 16px 0 rgba(${primaryRgb.r}, ${primaryRgb.g}, ${primaryRgb.b}, 1);
        `;
  }};
`;

export const Underline = styled.span`
  background-image: linear-gradient(
    120deg,
    ${({ theme }) => theme.underlineStart} 0%,
    ${({ theme }) => theme.underlineEnd} 100%
  );
  background-repeat: no-repeat;
  background-size: 100% 0.2em;
  background-position: 0 105%;
  transition: background-size 0.25 ease-in;

  &:hover {
    background-size: 110% 98%;
    color: ${({ theme }) => theme.body};
  }
`;
