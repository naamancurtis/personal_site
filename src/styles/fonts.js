// import { createGlobalStyle} from 'styled-components';

// import light from '../assets/fonts/FiraSansExtraCondensed-Light.ttf';
// import regular from '../assets/fonts/FiraSansExtraCondensed-Regular.ttf';
// import bold from '../assets/fonts/FiraSansExtraCondensed-SemiBold.ttf';

// export default createGlobalStyle`
//   @font-face {
//     font-family: 'Fira Sans';
//     src: local('Fira Sans Extra Condensed'),
//       url(${light}) format('ttf');
//     font-weight: 300;
//     font-style: normal;
//   };

//   @font-face {
//     font-family: 'Fira Sans';
//     src: local('Fira Sans Extra Condensed'),
//       url(${regular}) format('ttf');
//     font-weight: 400;
//     font-style: normal;
//   };

//   @font-face {
//     font-family: 'Fira Sans';
//     src: local('Fira Sans Extra Condensed'),
//       url(${bold}) format('ttf');
//     font-weight: 600;
//     font-style: normal;
//   };
// `
