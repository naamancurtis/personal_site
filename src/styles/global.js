import { createGlobalStyle } from 'styled-components';

export default createGlobalStyle`
  *,
  *::after,
  *::before {
    box-sizing: border-box; 
  }

  body {
    align-items: center;
    background: ${({ theme }) => theme.body};
    color: ${({ theme }) => theme.text};
    display: flex;
    flex-direction: column;
    justify-content: center;
    height: 100vh;
    margin: 0;
    padding: 0;
    font-family: 'Fira Sans Extra Condensed', Roboto, Helvetica;
    transition: all 0.25s linear;
  }
`;
