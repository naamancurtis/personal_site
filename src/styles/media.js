import { css } from 'styled-components';

const sizes = {
  uhd: 1980,
  widescreen: 1366,
  desktop: 1024,
  tablet: 768,
};

// Used for react-media - grabbing the size within components
export const GLOBAL_MEDIA_QUERIES = {
  uhd: '(min-width: 1980px)',
  widescreen: '(min-width: 1366px) and (max-width: 1979px)',
  desktop: '(min-width: 1024px) and (max-width: 1365px)',
  tablet: '(min-width: 768px) and (max-width: 1023px)',
  mobile: '(max-width: 767px)',
};

// Used for styled components
export default Object.keys(sizes).reduce((acc, label) => {
  acc[label] = (...args) => css`
    @media (min-width: ${sizes[label]}px) {
      ${css(...args)};
    }
  `;

  return acc;
}, {});
