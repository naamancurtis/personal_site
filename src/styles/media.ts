import {
  css,
  CSSObject,
  FlattenSimpleInterpolation,
  SimpleInterpolation,
} from 'styled-components';

type ObjectMap<T> = { [key: string]: T };

const sizes: ObjectMap<number> = {
  uhd: 1980,
  widescreen: 1366,
  desktop: 1024,
  tablet: 768,
};

// Used for react-media - grabbing the size within components
export const GLOBAL_MEDIA_QUERIES = {
  uhd: '(min-width: 1980px)',
  widescreen: '(min-width: 1366px) and (max-width: 1979px)',
  desktop: '(min-width: 1024px) and (max-width: 1365px)',
  tablet: '(min-width: 768px) and (max-width: 1023px)',
  mobile: '(max-width: 767px)',
};

// Used for styled components
export default Object.keys(sizes).reduce(
  (
    acc: {
      [key: string]: (
        first: TemplateStringsArray | CSSObject,
        ...interpolations: SimpleInterpolation[]
      ) => FlattenSimpleInterpolation;
    },
    label
  ) => {
    acc[label] = (...args) => css`
      @media (min-width: ${sizes[label]}px) {
        ${css(...args)};
      }
    `;

    return acc;
  },
  {}
);
