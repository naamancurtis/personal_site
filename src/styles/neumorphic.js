import { css } from 'styled-components';
import { pSBC, convertRgbStrToObject } from './utility.functions';

export const defaultBoxShadowCss = css`
  box-shadow: ${({ theme }) => {
    let topLeft;
    let bottomRight;
    let topOpacity;
    let bottomOpacity;

    if (theme.theme === 'dark') {
      // Dark Mode
      bottomRight = convertRgbStrToObject(pSBC(-0.3, theme.body, 'c'));
      topLeft = convertRgbStrToObject(pSBC(0, theme.contrastBody, 'c'));
      topOpacity = 0.1;
      bottomOpacity = 0.9;
    } else {
      // Light Mode
      topLeft = convertRgbStrToObject(pSBC(0.5, theme.body, 'c'));
      bottomRight = convertRgbStrToObject(pSBC(0.1, theme.contrastBody, 'c'));
      topOpacity = 1;
      bottomOpacity = 0.15;
    }

    return `
        -3px -3px 8px 0 rgba(${topLeft.r}, ${topLeft.g}, ${topLeft.b}, ${topOpacity}),
        6px 6px 10px 0 rgba(${bottomRight.r}, ${bottomRight.g}, ${bottomRight.b}, ${bottomOpacity});
        `;
  }};
`;

export const emphasisedBoxShadowCss = css`
  box-shadow: ${({ theme }) => {
    let topLeft;
    let bottomRight;
    let topOpacity;
    let bottomOpacity;

    if (theme.theme === 'dark') {
      // Dark Mode
      bottomRight = convertRgbStrToObject(pSBC(-0.3, theme.body, 'c'));
      topLeft = convertRgbStrToObject(pSBC(0, theme.contrastBody, 'c'));
      topOpacity = 0.1;
      bottomOpacity = 0.7;
    } else {
      // Light Mode
      topLeft = convertRgbStrToObject(pSBC(0.5, theme.body, 'c'));
      bottomRight = convertRgbStrToObject(pSBC(0.1, theme.contrastBody, 'c'));
      topOpacity = 1;
      bottomOpacity = 0.15;
    }

    return `
        -3px -3px 3px 0 rgba(${topLeft.r}, ${topLeft.g}, ${topLeft.b}, ${topOpacity}),
        6px 6px 2px 0 rgba(${bottomRight.r}, ${bottomRight.g}, ${bottomRight.b}, ${bottomOpacity});
        `;
  }};
`;

export const indentBoxShadowCss = css`
  box-shadow: ${({ theme }) => {
    let topLeft;
    let bottomRight;
    let topOpacity;
    let bottomOpacity;

    if (theme.theme === 'dark') {
      // Dark Mode
      bottomRight = convertRgbStrToObject(pSBC(-0.3, theme.body, 'c'));
      topLeft = convertRgbStrToObject(pSBC(0, theme.contrastBody, 'c'));
      topOpacity = 0.07;
      bottomOpacity = 0.8;
    } else {
      // Light Mode
      topLeft = convertRgbStrToObject(pSBC(0.5, theme.body, 'c'));
      bottomRight = convertRgbStrToObject(pSBC(0.1, theme.contrastBody, 'c'));
      topOpacity = 1;
      bottomOpacity = 0.15;
    }

    return `
        inset -6px -6px 8px 0 rgba(${topLeft.r}, ${topLeft.g}, ${topLeft.b}, ${topOpacity}),
        inset 3px 3px 10px 0 rgba(${bottomRight.r}, ${bottomRight.g}, ${bottomRight.b}, ${bottomOpacity});
        `;
  }};
`;

export const clickedBoxShadowCss = css`
  box-shadow: ${({ theme }) => {
    let topLeft;
    let bottomRight;
    let topOpacity;
    let bottomOpacity;

    if (theme.theme === 'dark') {
      // Dark Mode
      bottomRight = convertRgbStrToObject(pSBC(-0.4, theme.body, 'c'));
      topLeft = convertRgbStrToObject(pSBC(0.1, theme.contrastBody, 'c'));
      topOpacity = 0.07;
      bottomOpacity = 0.8;
    } else {
      // Light Mode
      topLeft = convertRgbStrToObject(pSBC(0.6, theme.body, 'c'));
      bottomRight = convertRgbStrToObject(pSBC(0.3, theme.contrastBody, 'c'));
      topOpacity = 1;
      bottomOpacity = 0.3;
    }

    return `
        inset -6px -6px 6px 0 rgba(${topLeft.r}, ${topLeft.g}, ${topLeft.b}, ${topOpacity}),
        inset 3px 3px 2px 1px rgba(${bottomRight.r}, ${bottomRight.g}, ${bottomRight.b}, ${bottomOpacity});
        `;
  }};
`;

export const underlineHighlightBoxShadow = css`
  box-shadow: ${({ theme }) => {
    const reducedOpacityStart = convertRgbStrToObject(
      pSBC(0, theme.underlineStart, 'c')
    );
    const reducedOpacityEnd = convertRgbStrToObject(
      pSBC(0, theme.underlineEnd, 'c')
    );

    return `
        -1px -1px 8px 0 rgba(${reducedOpacityStart.r}, ${reducedOpacityStart.g}, ${reducedOpacityStart.b}, 0.3),
        4px 4px 14px 0 rgba(${reducedOpacityEnd.r}, ${reducedOpacityEnd.g}, ${reducedOpacityEnd.b}, 0.4);
        `;
  }};
`;

export const underlineHighlightTextShadow = css`
  text-shadow: ${({ theme }) => {
    const reducedOpacityStart = convertRgbStrToObject(
      pSBC(0, theme.underlineStart, 'c')
    );
    const reducedOpacityEnd = convertRgbStrToObject(
      pSBC(0, theme.underlineEnd, 'c')
    );

    return `
        4px 4px 6px rgba(${reducedOpacityStart.r}, ${reducedOpacityStart.g}, ${reducedOpacityStart.b}, 0.1),
        1px 1px 3px rgba(${reducedOpacityEnd.r}, ${reducedOpacityEnd.g}, ${reducedOpacityEnd.b}, 0.2);
        `;
  }};
`;
