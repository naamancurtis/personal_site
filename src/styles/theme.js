export var lightTheme = {};
export var darkTheme = {};

const fonts = {
  code: `Ubuntu,'Fira Code', source-code-pro, 'Operator Mono', monospace`,
  heading: 'Raleway, Ubuntu, Fira Sans Extra Condensed, Fira Sans',
  body: 'Raleway, Ubuntu, Fira Sans Extra Condensed, Fira Sans',
};

lightTheme = {
  fonts: { ...fonts },
  theme: 'light',
  body: '#EDECEE',
  text: '#4C5D75',

  // Text Highlighting
  type: '#5698F5',
  keyword: '#2d3436',
  trait: '#294975',
  fnDef: '#F57D6E',
  params: '#4478C2',
  return: '#4478C2',

  // underlineStart: '#41B3A3',
  // underlineEnd: '#8860D0',

  underlineStart: '#ECA677',
  underlineEnd: '#6D62DF',

  error: '#E63A57',
  success: '#188D4F',

  contrastBody: darkTheme.body,
};

darkTheme = {
  ...lightTheme,
  theme: 'dark',
  body: '#393A47',
  text: '#DEDBE2',

  // Text Highlighting
  type: '#41D2DA',
  keyword: '#F76B8E',
  trait: '#2FCAA3',
  fnDef: '#59B6E8',
  params: '#F991FF',
  return: '#F76B8E',

  success: '#20CB81',

  underlineStart: '#2FCAA3',
  underlineEnd: '#8860D0',

  contrastBody: '#DEDBE2',
};
