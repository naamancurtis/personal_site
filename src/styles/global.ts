import { createGlobalStyle } from 'styled-components';
import { Theme } from './theme';

export default createGlobalStyle<{ theme: Theme }>`
  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box; 
  }

  body {
    background: ${({ theme }) => theme.body};
    color: ${({ theme }) => theme.text};
    font-size: 16px;
    width: calc(100% - 2rem);
    height: 100vh;
    margin: 0 1rem;

    h1,h2,h3,h4,h5,h6 {
      font-family: ${({ theme }) => theme.fonts.heading};
    }

    p {
      font-family: ${({ theme }) => theme.fonts.body};
      font-weight: 500;
    }
}`;
