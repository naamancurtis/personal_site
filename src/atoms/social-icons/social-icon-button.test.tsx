import React from 'react';
import { shallow, mount } from 'enzyme';
import {
  GitHubIcon,
  LinkedInIcon,
  MailIcon,
} from './social-icon-button.component';
import { SocialWrapperWithLabel } from './social-icon-button.styles';

describe('<GitHubIcon />', () => {
  it('expect to render component', () => {
    const wrapper = shallow(<GitHubIcon />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<LinkedInIcon />', () => {
  it('expect to render component', () => {
    const wrapper = shallow(<LinkedInIcon />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<MailIcon />', () => {
  it('expect to render component', () => {
    const wrapper = shallow(<MailIcon openModal={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('expect to openModal', () => {
    const mockOpenModal = jest.fn(() => {});
    const wrapper = mount(<MailIcon openModal={mockOpenModal} />);
    wrapper.find(SocialWrapperWithLabel).simulate('click');
    expect(mockOpenModal).toHaveBeenCalled();
    expect(mockOpenModal).toHaveBeenCalledTimes(1);
  });
});
