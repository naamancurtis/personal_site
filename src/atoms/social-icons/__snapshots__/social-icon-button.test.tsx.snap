// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GitHubIcon /> expect to render component 1`] = `
<BaseIcon
  icon={
    Array [
      "fab",
      "github",
    ]
  }
  name="GitHub"
  url="https://github.com/naamancurtis"
/>
`;

exports[`<LinkedInIcon /> expect to render component 1`] = `
<BaseIcon
  icon={
    Array [
      "fab",
      "linkedin",
    ]
  }
  name="LinkedIn"
  url="https://www.linkedin.com/in/naamancurtis/"
/>
`;

exports[`<MailIcon /> expect to render component 1`] = `
<styled.div
  className="clickable"
  onClick={[Function]}
>
  <styled.a
    aria-label="Open Contact Me Modal"
    className="clickable"
    role="button"
  >
    <FontAwesomeIcon
      border={false}
      className="clickable"
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "fas",
          "envelope",
        ]
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </styled.a>
</styled.div>
`;
